staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller_reconcile.go

```diff
@@ -423,6 +438,26 @@ func (c *policyController) reconcilePolicyBinding(namespace, name string, bindin
 	return nil
 }
 
+func (c *policyController) reconcilePolicyStatus(namespace, name string, definition *v1alpha1.ValidatingAdmissionPolicy) error {
```

Comment:
this code lives inside the admission plugin implementation in the generic apiserver library, and policyController will be active in aggregated API server instances so they can observe and honor admission policies, but we don't expect non-kube-apiserver instances to run the type checking and update status on policy objects, or to necessarily have status update permissions on arbitrary ValidatingAdmissionPolicy objects, right?

what keeps this controller from trying and perpetually failing to update status from aggregated API server instances, returning an error from reconcile, and perpetually requeuing the policy for sync?

Reply:
@jpbetz I am not sure about the "controller running outside of kube-apiserver" situation. Can we work together to figure this out.

Reply:
For preventing requeuing the policy, not trying would be a valid strategy.

Reply:
Alternatives I'm aware of:
1. Drop the error from `UpdateStatus` (or send it to `utilruntime.HandleError`) to prevent reconcile from perpetually requeuing. This is not ideal. The non-kube-apiservers will make calls that can never succeed, and the kube-apiservers will not be returning the error from `UpdateStatus` to the controller like it should
2. Detect specifically the error that non-kube-apiservers are expected to receive when not authorized to update the status, and alternative (1) but only in this case. 
3. Try to somehow "wire off" the status reconciliation for non-kube-apiservers. This glanced around and it wasn't obvious to me if there was a clean way to do this.
4. Run this status controller in the kube-controller-manager. This is a big change and has lots of complications, especially around CEL versioning that I don't like.

Is (2) acceptable? There will be some needless calls, which I don't like, but it seems like it should be otherwise safe?

Reply:
I've implemented (2). Let me know how you think.

Reply:
Skipping requeues on forbidden looks good.

@liggitt I was wondering if we could also somehow set permissions to refuse status updates from non-kube-apiservers by default? I couldn't think of a simple way off the top of my head.

Reply:
Read access to the policy object is not accidental but granted. Don't grant the write access to outside kube-apiserver should do it.

Reply:
Aggregated API servers *may* not have write access to status, or they may. Avoiding thrashing if missing permissions seems good, but relying on forbidden errors to keep them from touching status when we don't even think they have the correct wiring to be running this evaluation seems pretty gorpy. If we really don't want the status writer running/living in aggregated servers, it seems weird to put it in the admission plugin impl and start it in those servers.

Reply:
Since, in future, the Status object will contain more information and part of it will be written by aggregated servers, kicking anything that write to status out of api server may not be necessary. I cannot find a good way to detect if current api server is kube-openapi either. Can we add a new config option to enable/disable type checking?

Reply:
Adding a config will confuse the user. Here is the new solution: the controller will do a check to determine whether it should update the status by checking if the controller has access to the compiled-in OpenAPI definitions, and if the definitions contain anything outside the core APIs, like Deployment. I've sent out the implementation and it seems to be working fine. Again, we don't want to remove this part out of the api server because when we sort out CRD versioning problem, we have good reason to let the api server to type check CRDs, in future releases. Please let me know if that addresses your concern.

Reply:
The implementation is updated yet again. Now only the kube-apiserver will wire in the schema resolver. If the controller does not have the schema resolver, type checking will be completely skipped.

Reply:
thanks, open a follow-up issue to consider moving the reconciler to a controller outside the admission plugin if we want to limit it to kube-apiserver

Reply:
created as https://github.com/kubernetes/kubernetes/issues/116563